# Earth Horizon calculation

![earth math graph](images/earth_dist.JPG)

In [1]:
import math

earth_radius = 3959  # mi
earth_circumference = earth_radius * 2 * math.pi  # 24875 mi

In [15]:
def min_visible_height(arc_distance):
    # Find angle in radians by dividing arc distance by circumference and mult by 2pi
    angle = (arc_distance / earth_circumference) * 2 * math.pi
    hyp = earth_radius / math.cos(angle)
    return hyp

In [31]:
# Max height above center of earth visible from 60 miles away
min_visible_height(51)

3959.3285147584206

So at 51 miles, you should be able to see something 0.33 miles high (3959.33 - 3959.0 [earth radius]). The sears tower is 1729 feet, which is 1729/5280 = 0.33 miles high. So at about 51 miles out, you should no longer be able to see the Sears tower, assuming you are at the level of the lake. Also, the [Mirage](https://en.wikipedia.org/wiki/Mirage) effect can also change this.

Double-check: according to https://en.wikipedia.org/wiki/Horizon:

For an observer standing at the top of the Burj Khalifa (828 metres (2,717 ft) in height), the horizon is at a distance of 103 kilometres (64 mi).

In [30]:
# Check against  
min_visible_height(64)

3959.5173586827823

So at 64 miles, the minimum height you could see is 0.51 miles, which is 2692 feet. The Burj Khalifa is 2717 feet, which checks out.